src/transformers/pipelines/__init__.py

```diff
@@ -437,7 +437,7 @@ def pipeline(
 
     model_config = model.config
 
-    load_tokenizer = type(model_config) in TOKENIZER_MAPPING
+    load_tokenizer = type(model_config) in TOKENIZER_MAPPING or model_config.tokenizer_class is not None
```

Comment:
I see. I understand then why we want to have different code, however I feel like this does not really belong in pipelines.
Would `load_tokenizer = isinstance(model_config, Config)` be enough ? (So that the `model_config.tokenizer_class` is delegated to the `from_pretrained`) method.

Reply:
Hmm I'm not seeing 100% why this would be cleaner to  be honest. I don't see the relation between a model config being of type `PreTrainedConfig` and the tokenizer

Reply:
Otherwise what we could do is to add a `or tokenizer is not None`?